src/pulls.ts

```diff
@@ -16,8 +16,10 @@ export async function gatherPullRequest(
   maxRetries: number
 ): Promise<IGitHubPullRequest> {
   let tries = 0
-  let pullRequest: IGitHubPullRequest
-  let uknownStatus: boolean = typeof prEvent.pull_request.mergeable !== 'boolean'
+  // We query GitHub rightaway for merging state, as GitHub forces a compute of the mergeable state after a query
+  // See https://docs.github.com/en/rest/guides/getting-started-with-the-git-database-api#checking-mergeability-of-pull-requests
+  let pullRequest: IGitHubPullRequest = await getPullRequest(octokit, context, prEvent.number)
```

Comment:
This is incorrect since it will enter the loop and run a second time also it will run `maxTries +1` which would also be wrong